assignments/lambda-classes.js

```diff
@@ -1 +1,118 @@
 // CODE here for your Lambda Classes
+
+class Person {
+  constructor(props){
+    this.name = props.name;
+    this.age = props.age;
+    this.location = props.location;
+    this.gender = props.gender;
+  }
+  speak() {
+    // This method logs out a phrase Hello my name is Fred, I am from Bedrock where name and location are the object's own props
+    return console.log(`Hello my name is ${this.name}, I am from ${this.location}.`)
+  }
+}
+
+class Instructor extends Person {
+  constructor(props){
+    super(props);
+    this.specialty = props.specialty;
+    this.favLanguage = props.favLanguage;
+    this.catchPhrase = props.catchPhrase;
+  }
+  demo(subject) {
+    return console.log(`Today we are learning about ${subject}.`)
+  }
+  grade(student, subject) {
+    return console.log(`${student} receives a perfect score on ${subject}.`)
+  }
+  evaluate(student) {
+    student.grade += (Math.random() < 0.5 ? -1 : 1)*(Math.round(Math.random() * 10));
+    return student.grade;
+  }
+}
+
+class Student extends Person {
+  constructor(props){
+    super(props);
+    this.previousBackground = props.previousBackground;
+    this.className = props.className;
+    this.favSubjects = props.favSubjects;
+    this.grade = props.grade;
+  }
+  listsSubjects() {
+    return this.favSubjects.forEach(subject => {console.log(subject)})
+  }
+  PRAssignment(subject) {
+    return console.log(`${this.name} has submitted a PR for ${subject}.`)
+  }
+  sprintChallenge(subject) {
+    return console.log(`${this.name} has begun sprint challenge on ${subject}.`)
+  }
+}
+
+class ProjectManager extends Instructor {
+  constructor(props){
+    super(props);
+    this.gradClassName = props.gradClassName;
+    this.favInstructor = props.favInstructor;
+  }
+  standUp(channel) {
+    return console.log(`${this.name} announces to ${channel}, @channel standy times!`)
+  }
+  debugsCode(student, subject) {
+    return console.log(`${this.name} debugs ${student.name}'s code on ${subject}.`)
+  }
+}
+
+const fred = new Instructor({
+  name: 'Fred',
+  location: 'Bedrock',
+  age: 37,
+  gender: 'male',
+  favLanguage: 'StoneScript',
+  specialty: 'Flint-end',
+  catchPhrase: `Don't forget the homies`
+});
+
+const arnold = new Student({
+  name: 'Arnold',
+  location: 'Red Rock',
+  age: 26,
+  gender: 'male',
+  previousBackground: 'quarry yard manager',
+  className: 'CS-12',
+  favSubjects: [
+    'archaeology', 'paleontology', 'gerontology'
+  ],
+  grade: 75,
+})
+arnold.graduate = function() {return this.grade > 70 ? console.log(`${this.name} has graduated with a ${this.grade}!`) : console.log(`${this.name} did not pass and was re-evaluated to score ${fred.evaluate(arnold)}`)}
+arnold.speak = function(){}
```

Comment:
You probably want to add the graduate functionality to the Student class itself, but good job adding a method to an existing object!